distributed/tests/test_nanny.py

```diff
@@ -561,3 +563,24 @@ async def test_failure_during_worker_initialization(cleanup):
                 async with Nanny(s.address, foo="bar") as n:
                     await n
         assert "Restarting worker" not in logs.getvalue()
+
+
+def _exitcode_tester(_):
+    sleep(0.1)
```

Comment:
Is the sleep necessary? if not, I would suggest to remove it to reduce test runtime

Reply:
not needed :)

Reply:
It might be needed to make the test less flaky.

Reply:
I was hoping that the failed (windows) tests might get fixed by that.

I now let worker 0 die (instead of worker 1). Maybe the windows tests failed because all tasks were processed by worker 0?

Reply:
I think I know what is going wrong with windows: on Linux `gen_cluster(worker=Nanny)` creates a new process which can safely be killed with `sys.exit`, on windows it might just be a thread(?) in which case `sys.exit` might also kill the parent (speculating)?

Reply:
I honestly don't know and don't have access to a windows system right now to try this out. I do suspect that the signal is somehow treated differently. From what I saw in the tests, the logs where empty so the worker might not have been restarted?

Reply:
The reason why I assume that `sys.exit(1)` killed the entire pytest process is that the log file says at the bottom: "Error: Process completed with exit code 1."

Do we need a test case? Just changing `logger.info` to `logger.warn`. The `logger.info` wasn't covered byt a test.

Reply:
You can keep the test case but skip on windows. should be something like

```
from distributed.compatibility import WINDOWS
@pytest.mark.skipif(WINDOWS, reason="sys.exit(1) kills the entire pytest process on windows")
```